controllers/machine_controller.go

```diff
@@ -115,51 +138,66 @@ func (r *MachineReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
-func MakeSSHConfig(spec *naglfarv1.MachineSpec) *easyssh.MakeConfig {
-	timeout, _ := spec.Timeout.Parse()
-	return &easyssh.MakeConfig{
-		User:     spec.Username,
-		Password: spec.Password,
-		Server:   spec.Host,
-		Port:     strconv.Itoa(spec.SSHPort),
-		Timeout:  timeout,
+func (r *MachineReconciler) fetchMachineInfo(machine *naglfarv1.Machine, dockerClient docker.APIClient) (info *naglfarv1.MachineInfo, requeue bool, err error) {
+	if err = r.tryLock(machine, dockerClient); err != nil {
+		r.Eventer.Event(machine, "Warning", "Lock", err.Error())
+		return
 	}
-}
 
-func fetchMachineInfo(machine *naglfarv1.Machine) (*naglfarv1.MachineInfo, error) {
-	osStatScript, err := ScriptBox.FindString("os-stat.sh")
+	stat, err := dockerClient.ContainerInspect(r.Ctx, machineLock)
 
 	if err != nil {
-		return nil, err
+		r.Eventer.Event(machine, "Warning", "Check", err.Error())
+		return
 	}
 
-	ssh := MakeSSHConfig(&machine.Spec)
+	if timeIsZero(stat.State.StartedAt) {
+		err = dockerClient.ContainerStart(r.Ctx, machineLock, dockerTypes.ContainerStartOptions{})
+		requeue = true
+		return
+	}
 
-	stdout, stderr, done, err := ssh.Run(osStatScript)
+	if timeIsZero(stat.State.FinishedAt) {
+		requeue = true
+		return
+	}
 
-	if err != nil {
-		log.Error(err, "error in executing os-stat")
-		return nil, err
+	if stat.State.ExitCode != 0 {
+		err = fmt.Errorf("container exit with %d", stat.State.ExitCode)
+		r.Eventer.Event(machine, "Warning", "Fetch", err.Error())
+		return
 	}
 
-	if !done {
-		err = fmt.Errorf("script os-stat.sh not complete")
-		return nil, err
+	logs, err := dockerClient.ContainerLogs(r.Ctx, machineLock, dockerTypes.ContainerLogsOptions{
+		ShowStdout: true,
+		Follow:     true,
```

Comment:
It's no need to follow the log here because the container is finished already

Reply:
Fixed in [4f81fbd](https://github.com/PingCAP-QE/Naglfar/pull/58/commits/4f81fbd822fbb65766d752c5a3bb3c5ecea3fd76).